receiver/kafkametricsreceiver/consumer_scraper.go

```diff
@@ -0,0 +1,184 @@
+	topicPartitionOffset := map[string]map[int32]int64{}
+	for topic := range matchedTopics {
+		topicPartitionOffset[topic] = map[int32]int64{}
+		partitions, err := s.client.Partitions(topic)
+		if err != nil {
+			scrapeErrors.Add(err)
+			continue
+		}
+		for _, p := range partitions {
+			o, err := s.client.GetOffset(topic, p, sarama.OffsetNewest)
+			if err != nil {
+				scrapeErrors.Add(err)
+				continue
+			}
+			topicPartitions[topic] = append(topicPartitions[topic], p)
+			topicPartitionOffset[topic][p] = o
+		}
+	}
+	consumerGroups, listErr := s.clusterAdmin.DescribeConsumerGroups(matchedGrpIds)
+	if listErr != nil {
+		return metrics.ResourceMetrics(), listErr
+	}
+	for _, group := range consumerGroups {
+		grpMetrics := metrics.WithLabels(map[string]string{metadata.L.Group: group.GroupId})
+		grpMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupMembers.Name(), int64(len(group.Members)))
+		groupOffsetFetchResponse, err := s.clusterAdmin.ListConsumerGroupOffsets(group.GroupId, topicPartitions)
+		if err != nil {
+			scrapeErrors.Add(err)
+			continue
+		}
+		for topic, partitions := range groupOffsetFetchResponse.Blocks {
+			// tracking matchedTopics consumed by this group
+			// by checking if any of the blocks has an offset
+			isConsumed := false
+			for _, block := range partitions {
+				if block.Offset != -1 {
+					isConsumed = true
+					break
+				}
+			}
+			grpTopicMetrics := grpMetrics.WithLabels(map[string]string{metadata.L.Topic: topic})
+			if isConsumed {
+				var lagSum int64 = 0
+				var offsetSum int64 = 0
+				for partition, block := range partitions {
+					grpPartitionMetrics := grpTopicMetrics.WithLabels(map[string]string{metadata.L.Partition: string(partition)})
+					consumerOffset := block.Offset
+					offsetSum += consumerOffset
+					grpPartitionMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupOffset.Name(), consumerOffset)
+					// default -1 to indicate no lag measured.
+					var consumerLag int64 = -1
+					if partitionOffset, ok := topicPartitionOffset[topic][partition]; ok {
+						// only consider partitions with an offset
+						if block.Offset != -1 {
+							consumerLag = partitionOffset - consumerOffset
+							lagSum += consumerLag
+						}
+					}
+					grpPartitionMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupLag.Name(), consumerLag)
+				}
+				grpTopicMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupOffsetSum.Name(), offsetSum)
+				grpTopicMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupLagSum.Name(), lagSum)
+			}
+		}
+	}
+
+	return metrics.ResourceMetrics(), scrapeErrors.Combine()
+}
+
+func createConsumerScraper(_ context.Context, config Config, saramaConfig *sarama.Config, logger *zap.Logger) (scraperhelper.ResourceMetricsScraper, error) {
+	groupFilter, err := regexp.Compile(config.GroupMatch)
+	if err != nil {
+		return nil, err
+	}
+	topicFilter, err := regexp.Compile(config.TopicMatch)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't think these error messages will by default have enough context to know what particular regexp failed. You probably want to `return fmt.Errorf("failed to compile group_match: %w", err)` and so on. In general you want to make sure that an error message can be tied back to the line that emitted it by greping since you won't have a stack trace. So maybe consider adding to the ones below as well as any others throughout that don't have enough context.